starter-code/src/functions-and-arrays.js

```diff
@@ -74,6 +167,9 @@ var wordsCount = [
   'disobedience',
   'matter'
 ];
+
+/******************************************************/
+
 // Bonus Quest
 
```

Comment:
You can use this piece of code for the bonus quest :) 

function greatestProduct(matrix) {
  var big = 0;
  for (var i = 0; i < matrix.length; i++) {
    var left, right, up, down;
    var temp = 1;
    for (var j = 0; j < matrix.length; j++) {
      if (i === 0) {
        up = 1;
      } else if (i === matrix.length - 1) {
        down = 1;
      } else if (j === 0) {
        left = 1;
      } else if (j === matrix.length - 1) {
        right = 1;
      } else {
        left = matrix[i - 1][j];
        right = matrix[i + 1][j];
        up = matrix[i][j - 1];
        down = matrix[i][j + 1];
      }
      temp = left * right * up * down;
      if (temp > big) {
        big = temp;
      }
    }
  }
  return big;
}

// var matrix

var maxProduct = greatestProduct(matrix);
console.log(maxProduct);